src/translation.cpp

```diff
@@ -36,19 +37,28 @@ void Translations::clear()
 }
 
 const std::wstring &Translations::getTranslation(
-		const std::wstring &textdomain, const std::wstring &s)
+		const std::wstring &textdomain, const std::wstring &s) const
 {
 	std::wstring key = textdomain + L"|" + s;
-	try {
-		return m_translations.at(key);
-	} catch (const std::out_of_range &) {
-		verbosestream << "Translations: can't find translation for string \""
-		              << wide_to_utf8(s) << "\" in textdomain \""
-		              << wide_to_utf8(textdomain) << "\"" << std::endl;
-		// Silence that warning in the future
-		m_translations[key] = s;
-		return s;
+	auto it = m_translations.find(key);
+	if (it != m_translations.end())
+		return it->second;
+
+	// log warning once
+	thread_local std::vector<u64> logged_messages;
+	auto &log_to = tracestream;
+
+	if (log_to) {
+		u64 hash = murmur_hash_64_ua(key.data(), key.length(), 0x7366616E);
```

Comment:
~~This seed only works well in big endian.~~

Seeing how this was already degraded to tracestream, I wonder whether we need this log message at all. Even verbosestream is barely used (for debugging). This case of missing translations would rather be noticed by translators, seeing that it's incomplete in-game.

Reply:
I can also remove it entirely, that's even easier.